rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -0,0 +1,145 @@
+---
+Number: 0006
+Category: Standards Track
+Status: Draft
+Author: Ke Wang
+Organization: Nervos Foundation
+Created: 2018-12-01
+---
+
+# Complete Binary Merkle Tree
+
+## Introduction
+
+CKB uses Complete Binary Merkle Tree(CBMT) to generate Merkle Root and Merkle Proof for a static list of items, such as Transactions Root. Basically, CBMT is a complete binary tree as well as a full tree, it has the same node order with the complete binary tree. Compare with other Merkle trees, the hash computation of CBMT is minimal, as well as the proof size.
+
+## Tree Struct
+
+Tree with 6 items(the hashes are [T0, T1, T2, T3, T4, T5]) and tree with 7 items(the hashes are [T0, T1, T2, T3, T4, T5, T6]) is shown below:
```

Comment:
Why is 6 and 7 chosen here?

Reply:
just an example, 6 and 7 is friendly for drawing diagram.

Reply:
Please be explicit that 6 and 7 are trivial numbers. e.g. 

> For illustration purpose, we take trees with 6 and 7 leaves as example.